tests/rptest/tests/follower_fetching_test.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+import string
+from ducktape.mark import matrix
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+
+from rptest.services.cluster import cluster
+from rptest.services.kafka_cli_consumer import KafkaCliConsumer
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.services.redpanda import SISettings
+from rptest.tests.prealloc_nodes import PreallocNodesTest
+from rptest.util import wait_for_local_storage_truncate
+
+
+class FollowerFetchingTest(PreallocNodesTest):
+    def __init__(self, test_context):
+        self.log_segment_size = 1024 * 1024
+        self.local_retention = 2 * self.log_segment_size
+        si_settings = SISettings(
+            test_context,
+            cloud_storage_max_connections=5,
+            log_segment_size=self.log_segment_size,
+            cloud_storage_enable_remote_read=True,
+            cloud_storage_enable_remote_write=True,
+        )
+        self.s3_bucket_name = si_settings.cloud_storage_bucket
+
+        super(FollowerFetchingTest, self).__init__(test_context=test_context,
+                                                   num_brokers=3,
+                                                   node_prealloc_count=1,
+                                                   extra_rp_conf={
+                                                       'enable_rack_awareness':
+                                                       True,
+                                                   },
+                                                   si_settings=si_settings)
+
+    def tearDown(self):
+        self.cloud_storage_client.empty_bucket(self.s3_bucket_name)
+        super().tearDown()
+
+    def setUp(self):
+        # Delay startup, so that the test case can configure redpanda
+        # based on test parameters before starting it.
+        pass
+
+    def produce(self, topic, bytes=10 * 1024 * 1024):
+        msg_size = 512
+        msg_cnt = int(bytes / msg_size)
+        producer = KgoVerifierProducer(self.test_context, self.redpanda, topic,
+                                       msg_size, msg_cnt,
+                                       self.preallocated_nodes)
+        producer.start()
+        producer.wait()
+        producer.free()
+
+    def get_fetch_bytes(self, node, topic):
+        metrics = self.redpanda.metrics(node)
+        total_bytes = 0
+        for family in metrics:
+            for sample in family.samples:
+                if sample.name == "vectorized_cluster_partition_bytes_fetched_total" and sample.labels[
```

Comment:
was wondering if its useful to have a metric that accounts for total bytes fetched as a follower (non leader) to gauge the effectiveness of the feature in general.. (don't mean in this patch, just thinking out loud).

Reply:
May I ask also for metrics of client connection counts and bytes sent and received per rack ID?